File: design-pattern-jvm/app/src/test/java/design/pattern/java/patterns/creational/abstract_factory_method/AbstractFactoryMethodTest.java
package design.pattern.java.patterns.creational.abstract_factory_method;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class AbstractFactoryMethodTest {
  @Test
  public void itShouldCreateFordMustangWhenIsRequestedFromFordFactory() {
    VehicleFactory factory = new FordFactory();
    DeportVehicle car = factory.createDeportCar();
    Assertions.assertTrue(car instanceof FordMustang);
  }
  @Test
  public void itShouldCreateFordPickUpWhenIsRequestedFromFordFactory() {
    VehicleFactory factory = new FordFactory();
    RusticVehicle car = factory.createRusticCar();
    Assertions.assertTrue(car instanceof FordPickup);
  }
  @Test
  public void itShouldCreateToyotaYarisWhenIsRequestedFromToyotaFactory() {
    VehicleFactory factory = new ToyotaFactory();
    DeportVehicle car = factory.createDeportCar();
    Assertions.assertTrue(car instanceof ToyotaYaris);
  }
  @Test
  public void itShouldCreateForRunnerWhenIsRequestedFromToyotaFactory() {
    VehicleFactory factory = new ToyotaFactory();
    RusticVehicle car = factory.createRusticCar();
    Assertions.assertTrue(car instanceof ToyotaForRunner);
  }
}

File: design-pattern-jvm/app/src/test/java/design/pattern/kotlin/patterns/creational/abstract_factory_method/AbstractFactoryMethodTest.kt
package design.pattern.kotlin.patterns.creational.abstract_factory_method

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class AbstractFactoryMethodTest {
    @Test
    fun itShouldCreateFordMustangWhenIsRequestedFromFordFactory() {
        val factory:VehicleFactory = FordFactory()
        val car = factory.createDeportCar()
        Assertions.assertTrue(car is FordMustang)
    }

    @Test
    fun itShouldCreateFordPickUpWhenIsRequestedFromFordFactory() {
        val factory: VehicleFactory = FordFactory()
        val car = factory.createRusticCar()
        Assertions.assertTrue(car is FordPickup)
    }

    @Test
    fun itShouldCreateToyotaYarisWhenIsRequestedFromToyotaFactory() {
        val factory: VehicleFactory = ToyotaFactory()
        val car = factory.createDeportCar()
        Assertions.assertTrue(car is ToyotaYaris)
    }

    @Test
    fun itShouldCreateForRunnerWhenIsRequestedFromToyotaFactory() {
        val factory: VehicleFactory = ToyotaFactory()
        val car = factory.createRusticCar()
        Assertions.assertTrue(car is ToyotaForRunner)
    }
}